File: 2/2_cuatrimestre/PDOO/practicas/practica 1/deepSpaceRuby/ShieldBooster.rb
module Deepspace

    class ShieldBooster

        def initialize (name, boost, uses)
            @name = name
            @boost = boost
            @uses = uses
        end

        def self.newCopy (s)
            new(s.name, s.boost, s.uses)
        end

        def boost
            @boost
        end

        def uses
            @uses
        end

        def useIt
            if (@uses > 0)
                @uses -= 1
                @boost
            else
                1.0
            end
        end

    end

end

File: 2/2_cuatrimestre/PDOO/practicas/practica 1/deepSpace/src/deepspace/ShieldBooster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author manbolq
 */
public class ShieldBooster {
    private String name;
    private float boost;
    private int uses;
    
    
    /**************************************************************************/
    /*************************    CONSTRUCTORES   *****************************/
    /**************************************************************************/
    
    ShieldBooster(String name, float boost, int uses){
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster s){
        name = s.name;
        boost = s.boost;
        uses = s.uses;
    }
    
    
    /**************************************************************************/
    /**************************    CONSULTORES   ******************************/
    /**************************************************************************/
    
    float getBoost(){
        return boost;
    }
    
    int getUses(){
        return uses;
    }
    
    /**************************************************************************/
    /****************************    METODOS   ********************************/
    /**************************************************************************/
    
    public float useIt(){
        if (uses > 0){
            boost--;
            return boost;
        }
        else
            return 1.0f;
    }
}

